crypto/dh/dh_key.c

```diff
@@ -78,10 +78,15 @@ static int generate_key(DH *dh)
     int ok = 0;
     int generate_new_key = 0;
     unsigned l;
-    BN_CTX *ctx;
+    BN_CTX *ctx = NULL;
     BN_MONT_CTX *mont = NULL;
     BIGNUM *pub_key = NULL, *priv_key = NULL;
 
+    if (BN_num_bits(dh->p) > OPENSSL_DH_MAX_MODULUS_BITS) {
+        DHerr(DH_F_GENERATE_KEY, DH_R_MODULUS_TOO_LARGE);
+        goto err;
```

Comment:
Here I would just `return 0` rather than `goto err`. There's nothing to free yet, and below `err` we inject unnecessary generic error in addition to the more detailed error above.

Reply:
I agree with viktor.  Matt if you want to make that change when you merge, I approve.

Reply:
> I agree with viktor. 

Me too.